### PR TITLE
Create devtools_testing package for sharing integration tests

### PR DESCRIPTION
**Pull request by jakemac53**
Towards https://github.com/dart-lang/webdev/issues/633. Actually I think only flutter/flutter will use these though.

I was able to validate that given all these changes I could run tests using webdev/dwds by creating my own FlutterTestDriver and passing that in through my own FlutterTestEnvironment.

There is a lot to digest here but basically this PR:

- Creates a new devtools_testing package which can either be published on pub or depended on via a git constraint.
- Moves all the flutter integration tests and testing utilities in devtools into this new package
  - Each test gets its own entrypoint under lib, and a top level function that is identical to the old `main` except that it takes in a `FlutterTestEnvironment` argument.
  - The old test files in devtools now import and call these functions with the same environment as before.
- Moves all the test fixtures and goldens under the `devtools_testing` package, and updates the matchers to look for them in that package (they now have to live in a package so other packages can get them from pub).
  - The FlutterTestEnvironment figures out and sets the root devtools_testing package directory using a PackageResolver. This is a bit funky that it sets this up, but it basically has to be this way. The matchers can't do async code and resolving a package location is async.
 
Feel free to push back strongly - I know this seems a bit drastic but I think it is the easiest way to actually share the tests without copying them so they stay up to date.

### CODE REVIEW

File: packages/devtools_testing/pubspec.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+name: devtools_testing
+version: 1.0.0-dev
+environment:
+  sdk: ^2.2.0
```
Comment (jacob314) [inline packages/devtools_testing/pubspec.yaml]: nit: match devtools.

File: packages/devtools_testing/pubspec.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+dependencies:
```
Comment (jacob314) [inline packages/devtools_testing/pubspec.yaml]: add a comment with dependency overrides to use a relative import for package:devtools

**Comment by jakemac53** [comment]
cc @jonahwilliams as well for fyi

File: packages/devtools_testing/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+## 1.0.0-dev
```
[the remark this answers is not in the document]
Comment (jakemac53) [reply packages/devtools_testing/CHANGELOG.md]: Thankfully 2.0.0 is a thing :D

In all seriousness though we have gained absolutely nothing imo by not starting all packages as 1.0.0. It only causes confusion and arbitrary breakage just for the sake of going to 1.x.

File: packages/devtools_testing/pubspec.yaml (hunk 2 above)
Comment (jakemac53) [reply packages/devtools_testing/pubspec.yaml]: had to add a real override to resolve the merge conflict anyways :D

**Pull request merged**